build.gradle.kts

```
@@ -59,7 +59,7 @@ allprojects {
 subprojects {
     tasks.withType<Test>().configureEach {
         predictiveSelection {
-            enabled.set(System.getenv("CI") == null)
+            enabled.set(providers.gradleProperty("enablePTS").map(String::toBooleanStrict).orElse(false))
```

🥗 probably not necessary, because the default is listed in `gradle.properties` and if the property resolution fails or is empty because for example `-PenablePTS=` (or otherwise unset), that should be a fail.

```suggestion
            enabled.set(providers.gradleProperty("enablePTS").map(String::toBooleanStrict))
```

---

I added that exactly for that. If someone removes that line from `gradle.properties` I think it shouldn't fail.

---

But the point of `gradle.properties` is that it is the definition. No-one should be editing gradle.properties directly other than if you intend to change defaults. It's like removing a `val x = true` and then expecting that `foo(x)` will still compile :D

---

Also if we decide to change the default value, we have to change it twice now.

---

Agree, done